### PR TITLE
search contexts: add to extension API

### PR DESCRIPTION
**Pull request by novoselrok**
Added selected search context to the workspace and exposed it on the extension API.

### CODE REVIEW

File: client/extension-api/src/sourcegraph.d.ts [hunk 1]
```diff
@@ -1236,6 +1236,14 @@ declare module 'sourcegraph' {
...
          */
         export const versionContextChanges: Subscribable<string | undefined>
+
+        /**
+         * The current search context of the workspace, if any.
+         *
+         * A search context is a set of repositories and revisions on a Sourcegraph instance.
+         * When set, extensions use it to scope search queries, code intelligence actions, etc.
+         */
+        export const searchContext: string | undefined
```
Comment (tjkandala) [inline client/extension-api/src/sourcegraph.d.ts:1246]: Should we also add a way for extensions to be notified when the search context changes? Like `versionContextChanges`

**Comment by muratsu** [comment]
Hey @novoselrok, can you share docs for `search contexts` if there are any? I couldn't find anything about it on docs.sourcegraph.com. As we're exposing this API to 3rd party developers, we want to make sure we have links to relevant docs etc.

**Review by tjkandala — COMMENTED** [review]
Nice! Can you bump the version to 25.2.0 [here](https://sourcegraph.com/github.com/sourcegraph/sourcegraph/-/blob/client/extension-api/package.json#L3)?

**Comment by novoselrok** [comment]
> Hey @novoselrok, can you share docs for `search contexts` if there are any? I couldn't find anything about it on docs.sourcegraph.com. As we're exposing this API to 3rd party developers, we want to make sure we have links to relevant docs etc.

Good point @muratsu, we'll have to prepare the docs for the MVP features that currently exist. I'm converting this PR back to draft while we prepare the docs.

**Comment by novoselrok** [comment]
The docs were added here: https://docs.sourcegraph.com/code_search/explanations/features#search-contexts-experimental

**Review by tjkandala — APPROVED** [review]

**Pull request merged**